addons/stock_account/models/account_move.py

```diff
@@ -347,16 +348,12 @@ def _get_stock_layer_price_difference(self, layers, layers_price_unit, price_uni
             qty_to_correct = min(total_to_correct, remaining_qty)
             total_to_skip -= qty_to_skip
             total_to_correct -= qty_to_correct
-            layer_price_unit = self.company_id.currency_id._convert(
-                layers_price_unit[layer], po_line.currency_id, self.company_id, self.date, round=False)
-            price_difference = price_unit - layer_price_unit
-            price_difference = po_line.currency_id._convert(
-                price_difference, self.company_id.currency_id, self.company_id, self.date, round=False)
-            # TODO convert in invoice currency
-            price_difference_curr = (po_line.price_unit - self.price_unit)
-            if float_is_zero(price_difference * qty_to_correct, precision_rounding=self.currency_id.rounding):
+            unit_valuation_difference = price_unit - layers_price_unit[layer]
+            if float_is_zero(unit_valuation_difference * qty_to_correct, precision_rounding=self.company_id.currency_id.rounding):
                 continue
-            layers_to_correct[layer] = (qty_to_correct, price_difference, price_difference_curr)
+            po_pu_curr = po_line.currency_id._convert(po_line.price_unit, self.currency_id, self.company_id, self.date, round=False)
+            price_difference_curr = po_pu_curr - self.price_unit
```

Comment:
Its never == '0' ?

Reply:
It happens but if, therefore, we don't generate an SVL_pdiff, it would fail some tests. For instance this assert
https://github.com/odoo/odoo/blob/67c906f4d2ce5f150b77fe0f353e6688696318b0/addons/purchase_stock/tests/test_anglo_saxon_valuation_reconciliation.py#L178